redeployContract.js

```diff
@@ -211,31 +225,54 @@ const deployLargeContract = async (web3, target_address, contract_code, contract
         .on('receipt', function (receipt) {
             initContractAddress = receipt.contractAddress;
             console.log('Init address: ', receipt.contractAddress); // contains the new contract address
+            console.log(`Actual gas used: ${receipt.gasUsed}`);
             return receipt;
+        })
+        .catch(error => {
+            console.log(error);
+            process.exit();
         });
 
     initInstance.options.address = initContractAddress;
+
     // Set storage
+    console.log('Setting storage.');
     const storageKeys = Object.keys(contract_state);
+    let keysPerBatch = 2;
```

Comment:
Why is it initialized with 2? Doesn't this lead to issues if the state is empty?

Reply:
It's initialized with two, because in order to calculate the gas cost per additional transaction we need at least two key value pairs (calculate the difference between migrating one key value pair and two key value pairs)

Reply:
It does not lead to issues, because the function setValuesOnInitContract returns if the passed key array is empty. It only sends a transaction if there is at least on key value pair in the keys array.

Reply:
One problem that I see is that the code currently assumes that the first two key value pairs will always fit into one transaction..

Reply:
On second thought, I shouldve initialized with one.

Reply:
restructured the code and started calculcation at one